### PR TITLE
Enable Schul-Cloud integration

### PR DESCRIPTION
**Pull request by dominikglandorf**
Primarily a HTTP API configured via `data-`attributes for getting and sending data to a Schul-Cloud interface.

Additionally css hack for the add button and the `id` increment for using this great editor outside the small edtr.io dev universe.

Finally a small typo ;)

### CODE REVIEW

File: src/editor/App.js [hunk 1]
```diff
@@ -19,25 +19,44 @@ class App extends Component {
...
             <React.StrictMode>
-                <Provider store={this.state.store}>
+              { store
+                ? (
+                  <Provider store={this.state.store}>
                     <Editor />
-                </Provider>
+                  </Provider>
+                )
+                : <p>Lädt...</p>
```
Comment (FWirtz) [inline src/editor/App.js]: please replace `ä` with `&auml;`

Comment (FWirtz) [reply src/editor/App.js]: ...and `...` with `&hellip;`

**Comment by FWirtz** [comment]
I like the proposed changes.
This doesn't work though in the standalone version anymore.
I think we should support both versions to allow for easy development.

Let's talk about this tomorrow.

**Comment by dominikglandorf** [comment]
The add plugin button is now looking fabulous again. Also the stand alone version is working like a charm again!

File: src/common/api.js [hunk 2]
```diff
@@ -1,27 +1,57 @@
...
+          ? this.api.get(`/${this.worksheetId}`).then(result => {
+            return HttpApi.decode(result.data)
+          })
+          : Promise.resolve({})
+      );
+  }
+
+  static encode({doc, plugin}) {
+    return {
+      title: doc.title || "test",
```
Comment (FWirtz) [inline src/common/api.js]: Let's use e.g.  `"Untitled Worksheet"` instead of `"test"`

**Pull request merged**